README.md

```diff
@@ -1,277 +1,468 @@
-      value?: SDKNumber /* Transaction value */
-    }
-  ): Promise<{ txHash: string }> => {
+  sendTransaction: async (tx: {
+    from: EVMAddress; // Sender Ethereum address
+    to: EVMAddress; // Bridge Endpoint address
+    data: Uint8Array;
+    recommendedGasLimit: SDKNumber;
+    value?: SDKNumber;
+  }): Promise<{ txHash: string }> => {
     /**
      * Implementation for sending transaction on Ethereum mainnet
-     * See https://docs.ethers.org/v5/api/contract/contract/ for reference
+     * See https://docs.ethers.org/v6/ for reference
      */
     const txRequest = {
-    from: tx.from,
-    to: tx.to,
-    data: ethers.utils.hexlify(tx.data),
-    gasLimit: ethers.BigNumber.from(tx.recommendedGasLimit.split(" ")[0]), // Convert string to BigNumber
+      from: tx.from,
+      to: tx.to,
+      data: ethers.hexlify(tx.data),
+      // Convert SDKNumber into BigNumber
+      gasLimit: ethers.toBigInt(tx.recommendedGasLimit.split(" ")[0]), 
     };
 
     const sentTx = await signer.sendTransaction(txRequest);
     const receipt = await sentTx.wait();
-    return { txHash: receipt.transactionHash };
+    if (receipt === null) throw new Error("Transaction receipt is null");
+    return { txHash: receipt.hash };
   },
 };
-  
+
 // Perform the bridge operation
-const result = await broSdk.bridgeFromEVM(bridgeFromEVMInput);
-console.log("Transaction ID:", result.txHash);
+const result = await sdk.bridgeFromEVM(bridgeFromEVMInput);
+console.log("Ethereum Transaction ID:", result.txHash);
 ```
 
-#### Bridge from Bitcoin
+### Bridge From Bitcoin
+
+The following functions must be implemented by developer:
 
-Use case showcasing a transfer of 1 `BTC` from Bitcoin to `WBTC` on Ethereum using BroSDK.
+- `reselectSpendableUTXOs`: Selects UTXOs from the sender wallet.
+- `signPsbt`: Signs the PSBT (Partially Signed Bitcoin Transaction).
+- `sendTransaction`: Broadcasts the final transaction to the Bitcoin network.
 
-```typescript
+```ts
+import { BridgeFromBitcoinInput, KnownChainId, KnownTokenId } from "@brotocol-xyz/bro-sdk";
 import {
-  BridgeFromBitcoinInput,
-  KnownChainId,
-  KnownTokenId,
-  toSDKNumberOrUndefined,
-} from "@brotocol-xyz/bro-sdk"
-/* Use your preferred Bitcoin libs here */
-import { Psbt, networks, Transaction, script } from "bitcoinjs-lib";
+  GetConfirmedSpendableUTXOFn,
+  reselectSpendableUTXOsFactory,
+  UTXOBasic,
+} from "@brotocol-xyz/bro-sdk/bitcoinHelpers";
 import { ECPairFactory } from "ecpair";
-import * as tinysecp from "tiny-secp256k1";
+import { Psbt, payments, Signer, networks } from "bitcoinjs-lib";
 import axios from "axios";
+import { randomBytes } from "crypto";
+import * as ecc from "tiny-secp256k1";
+
+const rng = (size?: number) => {
+  const buffer = randomBytes(size || 0);
+  return Buffer.from(buffer);
+};
+
+// Create ECPair instance
+const ECPair = ECPairFactory(ecc);
+
+// Generate a new random key pair
+const keyPair = ECPair.makeRandom({ rng });
```

Comment:
I'd prefer we mock two functions: `getPublicKey(): Promise<Hex>` and `signTx(hex: Hex): Promise<Hex>` instead giving specific implementations, what would you guys think?